torque-core/src/test/kotlin/com/workday/torque/ChunkRetryerSpec.kt

```diff
@@ -0,0 +1,94 @@
+        val failedTestChunkResults = listOf(createPassedTest(adbDevice),
+                createFailedTest(adbDevice),
+                createPassedTest(adbDevice),
+                createFailedTest(adbDevice))
+        val passedTestChunkResults = listOf(createPassedTest(adbDevice),
+                createPassedTest(adbDevice),
+                createPassedTest(adbDevice),
+                createPassedTest(adbDevice))
+
+        given("A twice crashed then failed then passed test sequence") {
+            val testChunkRunner by memoized {
+                mockk<TestChunkRunner> {
+                    coEvery { run(args = any(), testChunk = any()) } returnsMany listOf(crashedTestChunkResults,
+                            crashedTestChunkResults,
+                            failedTestChunkResults,
+                            passedTestChunkResults)
+                }
+            }
+
+            on("Max retry count 3") {
+                val args = Args().apply {
+                    retriesPerChunk = 3
+                }
+                val chunkRetryer = ChunkRetryer(adbDevice, args, logcatFileIO, testChunkRunner, installer)
+
+                it("Retries until success and outputs test results with all passed tests") {
+                    runBlocking {
+                        val testResults = chunkRetryer.runTestChunkWithRetry(TestChunk(0, mockk(), mockk()))
+                        assertTrue {
+                            testResults.containsAll(passedTestChunkResults)
+                        }
+                    }
+                }
+            }
+
+            on("Max retry count 2") {
+                val args = Args().apply {
+                    retriesPerChunk = 2
+                }
+                val chunkRetryer = ChunkRetryer(adbDevice, args, logcatFileIO, testChunkRunner, installer)
+
+                it("Retries until max retries and outputs test results with failed tests") {
+                    runBlocking {
+                        val testResults = chunkRetryer.runTestChunkWithRetry(TestChunk(0, mockk(), mockk()))
+                        assertTrue {
+                            testResults.containsAll(failedTestChunkResults)
+                        }
+                    }
+                }
+            }
+
+            on("Max retry count 1") {
+                val args = Args().apply {
+                    retriesPerChunk = 1
+                }
+                val chunkRetryer = ChunkRetryer(adbDevice, args, logcatFileIO, testChunkRunner, installer)
+
+                it("Retries until max retries and outputs test results with failed crashed tests") {
+                    runBlocking {
+                        val testResults = chunkRetryer.runTestChunkWithRetry(TestChunk(0, mockk(), mockk()))
+                        assertTrue {
+                            testResults.containsAll(crashedTestChunkResults)
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Can you add a test for the case where there are no retries allowed? (so we would expect this to behave the same as the regular TestChunkRunner)